ibis_vega_transform/vegaexpr.py

```diff
@@ -112,126 +162,161 @@ def now() -> float:
     return ibis.now()
 
 
-def datetime(
-    year: int,
-    month: int,
-    day: int = 0,
-    hour: int = 0,
-    min: int = 0,
-    sec: int = 0,
-    millisec: int = 0,
-) -> dt.datetime:
-    """Returns a new Date instance.
-    The month is 0-based, such that 1 represents February.
+def z(v) -> bool:
     """
-    # TODO: do we need a local timezone?
-    return dt.datetime(
-        int(year),
-        int(month) + 1,
-        int(day),
-        int(hour),
-        int(min),
-        int(sec),
-        int(millisec * 1000),
-    )
+    Returns whether the value is equal to a literal 0
+    """
+    return not isinstance(v, ibis.expr.types.Expr) and v == 0
 
 
-def date(datetime: dt.datetime) -> int:
+def datetime(y, m, d, h, mi, s, ms):
+    """
+    This should use https://github.com/ibis-project/ibis/issues/386 if it gets implemented
+
+    Currently we only support use cases that can be turned into a truncate, which is:
+
+        datetime(year(col), 0, 0, 0, 0, 0)
+        or
+        datetime(year(col), month(col), 0, 0, 0, 0)
+        etc.
+    """
+    if not hasattr(y, "_arg"):
+        raise NotImplementedError()
+    if not isinstance(y._arg, ibis.expr.operations.ExtractYear):
+        raise NotImplementedError()
+    date_column = y._arg.arg
```

Comment:
Yikes, where does `_arg` come from?

Reply:
Who knows? I just found it on here. @xmnlab do you know of a better way to implement this sort of logic? I am trying to take apart an ibis expression into it's pieces to see how it was created.